palette/src/gradient.rs

```diff
@@ -439,7 +447,7 @@ mod test {
         ]);
         let g2 = g1.slice(..0.5);
 
-        let v1: Vec<_> = g1.take(10).take(5).collect();
+        let v1: Vec<_> = g1.take(9).take(5).collect();
```

Comment:
9 items is 8 "jumps" and 5 items is 4 "jumps". 5 ends at the middle of 9. It's slightly mind bending (for my post work Friday brain), but it checks out.

Reply:
Yes, I had to use pen and paper to reason about it. My thinking was the interval between each color step is 1/(n-1) for n > 1. So 9 is 1/8, 10 is 1/9, etc. 

When you take 9, you're getting 7 intermediate colors that are 8 "jumps" apart. At first, this can be confusing when reasoning about what to take if you want a specific interval. I wanted to add a short but hopefully clear explanation of this in the docs so future people can avoid bending minds.